src/dato/lib/server.clj

```diff
@@ -325,33 +282,49 @@
 (defn new-routing-table [table]
   (merge default-routing-table table))
 
-(defn new-session-id []
-  (str (UUID/randomUUID)))
-
-(defrecord DatoServer [routing-table datomic-uri]  )
-
-(defn ?assign-id [handler dato-server]
-  (fn [request]
-    (let [id        (get-in request [:session :id] (new-session-id))
-          mouse-pos [0 0]
-          response  (handler (-> request
-                                 (assoc-in [:session :id] id)
-                                 (assoc-in [:session :mouse :pos] mouse-pos)
-                                 (assoc-in [:session :live :dato] dato-server)))]
-      response)))
+(defrecord DatoServer [routing-table datomic-uri])
+
+(defn validate-config [config]
+  (assert (:server config) "config must have a server field")
+  (assert (:port config) "config must have a port field"))
+
+(defn wrap-server-routes [tal-state]
+  (let [ws-setup (tal-routes/websocket-setup tal-state)
+        ajax-poll (tal-routes/ajax-poll tal-state)
+        ajax-send (tal-routes/ajax-send tal-state)]
+    (fn [req]
+      (case (:uri req)
+        "/talaria" (ws-setup req)
+        "/talaria/ajax-poll" (ajax-poll req)
+        "/talaria/ajax-send" (ajax-send req)
+        nil))))
+
+(defn set-dato-route-heirarchy!
```

Comment:
Very unclear on how this works, bit nervous about that. It's ok for now, but something I'll probably want to think about long term.